hw/ip/dma/doc/programmers_guide.md

```diff
@@ -0,0 +1,93 @@
+The following three types of interrupts are supported:
+
+1.  **Transfer Completion:** An interrupt is raised when the entire data transfer (as defined by `TOTAL_DATA_SIZE`) is complete.
+2.  **Chunk Completion:** An interrupt is raised after the transfer of a single chunk of data (as defined by `CHUNK_DATA_SIZE`) is finished.
+3.  **Error Condition:** An interrupt is raised if an error occurs during the transfer process.
+
+The current status of the DMA, including pending interrupts, can be read from the [`STATUS`](registers.md#status) register.
+Since interrupts are implemented as status bits, they are cleared by writing a '1' to the corresponding bit in the `STATUS` register.
+
+### Aborting a Transfer
+
+Software can terminate an ongoing DMA transfer at any point by writing to the `abort` bit in the [`CONTROL`](registers.md#control) register.
+
+## Chunked Data Transfers
+
+The DMA performs memory transfers in discrete units called chunks.
+Each chunk consists of a contiguous block of data with a size defined by the [`CHUNK_DATA_SIZE`](#chunk_data_size) register.
+After transferring a chunk, the DMA can optionally generate an interrupt.
+
+While chunked transfers are primarily utilized in conjunction with the [Hardware Handshaking Mode](#Hardware_Handshaking_Mode) for interacting with IO peripherals, they can also be employed in memory-to-memory transfers.
+A potential use case for chunked memory-to-memory transfers is memory initialization, where a small chunk of data (e.g., a block of zeros) can be repeatedly transferred to a larger memory region.
+
+## Hardware Handshaking Mode
+
+The DMA supports a hardware handshaking mode that enables seamless data transfers between IO peripherals and memory.
+This mode leverages the chunked data transfer mechanism and interrupts from the IO peripheral.
+
+In this mode, the IO peripheral fills its internal transfer FIFO and then signals the DMA by raising an interrupt.
+Upon receiving this interrupt, the DMA reads the data from the peripheral's FIFO and writes it to the destination memory.
+This process continues in a loop: the DMA transfers a chunk of data, waits for the IO peripheral to fill its FIFO and issue another interrupt, and then transfers the next chunk.
+This loop repeats until the total number of bytes specified by [`TOTAL_DATA_SIZE`](registers.md#total_data_size) has been transferred.
+
+To enable hardware handshaking, software must set the `hardware_handshake_enable` bit in the [`CONTROL`](#control) register and configure the appropriate addressing mode to access the IO peripheral.
+
+The DMA's chunked transfer mechanism in hardware handshaking mode supports two address update modes for the peripheral:
+
+  * **Continuous Access:** The DMA continuously accesses the same address for each chunk.
+  * **Wrap-around:** After transferring a chunk, the DMA's peripheral address wraps around to the starting address configured for the transfer.
```

Comment:
I think the description here could match up better with the register bit descriptions in the hjson. The latest incarnation of the DMA has two independent 'addressing bits' doesn't it; determining (i) what happens after each (partial-)word transfer and (ii) what happens at the end of each chunk?